core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -408,7 +408,9 @@ static TEE_Result tee_ta_load_check_shdr(struct shdr *shdr)
 				SHDR_GET_SIG(shdr), shdr->sig_size);
 out:
 	crypto_ops.acipher.free_rsa_public_key(&key);
-	return res;
+	if (res != TEE_SUCCESS)
+		return TEE_ERROR_SECURITY;
+	return TEE_SUCCESS;
```

Comment:
I guess an exception for that would be OK since the return code doesn't depend on the supplied TA.
Should I update the commit message instead or which do you prefer?